hail/src/main/scala/is/hail/io/IndexBTree.scala

```diff
@@ -89,18 +88,18 @@ object IndexBTree {
     btreeLayers(arr, branchingFactor).map(_.mkString("[", " ", "]")).mkString("(BTREE\n", "\n", "\n)")
 }
 
-class IndexBTree(indexFileName: String, hConf: Configuration, branchingFactor: Int = 1024) extends Closeable {
+class IndexBTree(indexFileName: String, fs: FS, branchingFactor: Int = 1024) extends Closeable {
   val maxDepth = calcDepth()
-  private val fs = try {
-    hConf.fileSystem(indexFileName).open(new Path(indexFileName))
+  private val fileSystem = try {
+    fs.fileSystem(indexFileName).open(indexFileName)
```

Comment:
Could be more simply .open in the new implementation if desired. This seemed cleaner (although uglier code) from a PR standpoint.

Reply:
This feels like it should be `open`.